File: VB/SpreadsheetDocServerPivotAPI/CodeExamples/PivotTableFormattingActions.vb
Imports DevExpress.Spreadsheet
Imports System

Namespace SpreadsheetDocServerPivotAPI
	Public Module PivotTableFormattingActions

		Public ChangeStylePivotTableAction As Action(Of Workbook) = AddressOf ChangeStylePivotTable
		Public BandedColumnsAction As Action(Of Workbook) = AddressOf BandedColumns
		Public BandedRowsAction As Action(Of Workbook) = AddressOf BandedRows
		Public ShowColumnHeadersAction As Action(Of Workbook) = AddressOf ShowColumnHeaders
		Public ShowRowHeadersAction As Action(Of Workbook) = AddressOf ShowRowHeaders


		Private Sub ChangeStylePivotTable(ByVal workbook As IWorkbook)
#Region "#SetStyle"
			Dim worksheet As Worksheet = workbook.Worksheets("Report1")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Access the pivot table by its name in the collection.
			Dim pivotTable As PivotTable = worksheet.PivotTables("PivotTable1")
			' Set the pivot table style.
			pivotTable.Style = workbook.TableStyles(BuiltInPivotStyleId.PivotStyleDark7)
#End Region ' #SetStyle
		End Sub


		Private Sub BandedColumns(ByVal workbook As IWorkbook)
#Region "#BandedColumns"
			Dim worksheet As Worksheet = workbook.Worksheets("Report4")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Access the pivot table by its name in the collection.
			Dim pivotTable As PivotTable = worksheet.PivotTables("PivotTable1")
			' Apply the banded column formatting to the pivot table. 
			pivotTable.BandedColumns = True
#End Region ' #BandedColumns
		End Sub

		Private Sub BandedRows(ByVal workbook As IWorkbook)
#Region "#BandedRows"
			Dim worksheet As Worksheet = workbook.Worksheets("Report4")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Access the pivot table by its name in the collection.
			Dim pivotTable As PivotTable = worksheet.PivotTables("PivotTable1")
			' Apply the banded row formatting to the pivot table. 
			pivotTable.BandedRows = True
#End Region ' #BandedRows
		End Sub

		Private Sub ShowColumnHeaders(ByVal workbook As IWorkbook)
#Region "#ColumnHeaders"
			Dim worksheet As Worksheet = workbook.Worksheets("Report1")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Access the pivot table by its name in the collection.
			Dim pivotTable As PivotTable = worksheet.PivotTables("PivotTable1")
			' Remove formatting from column headers.
			pivotTable.ShowColumnHeaders = False
#End Region ' #ColumnHeaders
		End Sub

		Private Sub ShowRowHeaders(ByVal workbook As IWorkbook)
#Region "#RowHeaders"
			Dim worksheet As Worksheet = workbook.Worksheets("Report1")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Access the pivot table by its name in the collection.
			Dim pivotTable As PivotTable = worksheet.PivotTables("PivotTable1")
			' Add the "Region" field to the column axis area. 
			pivotTable.ColumnFields.Add(pivotTable.Fields("Region"))
			' Remove formatting from row headers.
			pivotTable.ShowRowHeaders = False
#End Region ' #RowHeaders
		End Sub
	End Module
End Namespace

File: CS/SpreadsheetDocServerPivotAPI/CodeExamples/PivotTableFormattingActions.cs
using DevExpress.Spreadsheet;

namespace SpreadsheetDocServerPivotAPI
{
    public static class PivotTableFormattingActions
    {

        static void ChangeStylePivotTable(IWorkbook workbook)
        {
            #region #SetStyle
            Worksheet worksheet = workbook.Worksheets["Report1"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Access the pivot table by its name in the collection.
            PivotTable pivotTable = worksheet.PivotTables["PivotTable1"];
            // Set the pivot table style.
            pivotTable.Style = workbook.TableStyles[BuiltInPivotStyleId.PivotStyleDark7];
            #endregion #SetStyle
        }


        static void BandedColumns(IWorkbook workbook)
        {
            #region #BandedColumns
            Worksheet worksheet = workbook.Worksheets["Report4"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Access the pivot table by its name in the collection.
            PivotTable pivotTable = worksheet.PivotTables["PivotTable1"];
            // Apply the banded column formatting to the pivot table. 
            pivotTable.BandedColumns = true;
            #endregion #BandedColumns
        }

        static void BandedRows(IWorkbook workbook)
        {
            #region #BandedRows
            Worksheet worksheet = workbook.Worksheets["Report4"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Access the pivot table by its name in the collection.
            PivotTable pivotTable = worksheet.PivotTables["PivotTable1"];
            // Apply the banded row formatting to the pivot table. 
            pivotTable.BandedRows = true;
            #endregion #BandedRows
        }

        static void ShowColumnHeaders(IWorkbook workbook)
        {
            #region #ColumnHeaders
            Worksheet worksheet = workbook.Worksheets["Report1"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Access the pivot table by its name in the collection.
            PivotTable pivotTable = worksheet.PivotTables["PivotTable1"];
            // Remove formatting from column headers.
            pivotTable.ShowColumnHeaders = false;
            #endregion #ColumnHeaders
        }

        static void ShowRowHeaders(IWorkbook workbook)
        {
            #region #RowHeaders
            Worksheet worksheet = workbook.Worksheets["Report1"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Access the pivot table by its name in the collection.
            PivotTable pivotTable = worksheet.PivotTables["PivotTable1"];
            // Add the "Region" field to the column axis area. 
            pivotTable.ColumnFields.Add(pivotTable.Fields["Region"]);
            // Remove formatting from row headers.
            pivotTable.ShowRowHeaders = false;
            #endregion #RowHeaders
        }
    }
}
